### PR TITLE
[socker.io] initialization options inherit from engine.io

### PR DESCRIPTION
**Pull request by KentarouTakeda**
Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  
  * https://github.com/socketio/socket.io/blob/master/docs/API.md#new-serverhttpserver-options
    > The same options passed to socket.io are always passed to the engine.io Server that gets created. See engine.io options as reference.
  * https://github.com/socketio/socket.io/blob/2.2.0/lib/index.js#L310
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

This improvement include fix of #44788

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## engine.io ([<kbd>unpkg</kbd>](https://unpkg.com/browse/engine.io@latest/))
was missing the following properties:
1. Transport
1. transports
1. parser
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c7e2c76815cb0d8738d12d79b1427aa4282d80cc
</p>

File: types/engine.io/index.d.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ declare namespace engine {
          * conform to the ws interface (see ws module api docs). Default value is ws.
          * An alternative c++ addon is also available by installing uws module.
          */
-        wsEngine?: "ws"|"uws";
+        wsEngine?: "ws"|"uws"|string;
```
Comment (murbanowicz) [inline types/engine.io/index.d.ts:85]: Can you remove this change? I do not see this necessary and on top of that it blocks us from quick fixing the issue.

**Review by murbanowicz — CHANGES_REQUESTED** [review]

File: types/engine.io/index.d.ts (hunk 1 above)
Comment (KentarouTakeda) [reply types/engine.io/index.d.ts]: Fixed.

The fix in the original issue was `string`, so I followed it, but when I checked the code in engine.io, it certainly accepted only `ws`, `uws`. Thank you.

https://github.com/socketio/engine.io/blob/3.4.1/lib/server.js#L109

I wrote the code in the past but forgot it :)

**Review by murbanowicz — APPROVED** [review]

**Comment by murbanowicz** [comment]
> @KentarouTakeda One or more reviewers has requested changes. Please address their comments. I'll be back once they sign off or you've pushed new commits or comments. If you disagree with the reviewer's comments, you can "dismiss" the review using GitHub's review UI. Thank you!

That is fine. That's why we have code review :)

**Comment by KentarouTakeda** [comment]
Ready to merge

**Pull request merged**